packages/core/src/plugins/manifest.ts

```diff
@@ -0,0 +1,154 @@
+  assets: FilePath[];
+  html?: FilePath;
+};
+
+type ManifestList = {
+  entries: {
+    /** relate to rsbuild source.entry */
+    [entryName: string]: ManifestByEntry;
+  };
+  /** Flatten all assets */
+  allFiles: FilePath[];
+};
+
+type FileDescriptor = {
+  chunk?: Chunk;
+  isAsset: boolean;
+  isChunk: boolean;
+  isInitial: boolean;
+  isModuleAsset: boolean;
+  name: string;
+  path: string;
+};
+
+export const generateManifest =
+  (htmlPaths: Record<string, string>) =>
+  (_seed: Record<string, any>, files: FileDescriptor[]) => {
+    const chunkEntries = new Map<string, FileDescriptor[]>();
+
+    const licenseMap = new Map<string, string>();
+
+    const allFiles = files.map((file) => {
+      if (file.chunk) {
+        const names = recursiveChunkEntryNames(file.chunk);
+
+        for (const name of names) {
+          chunkEntries.set(name, [file, ...(chunkEntries.get(name) || [])]);
+        }
+      }
+
+      if (file.path.endsWith('.LICENSE.txt')) {
+        const sourceFilePath = file.path.split('.LICENSE.txt')[0];
+        licenseMap.set(sourceFilePath, file.path);
+      }
+      return file.path;
+    });
+
+    const entries: ManifestList['entries'] = {};
+
+    for (const [name, chunkFiles] of chunkEntries) {
+      const assets = new Set<string>();
+      const initialJS: string[] = [];
+      const asyncJS: string[] = [];
+      const initialCSS: string[] = [];
+      const asyncCSS: string[] = [];
+
+      for (const file of chunkFiles) {
+        if (file.isInitial) {
+          if (file.path.endsWith('.css')) {
+            initialCSS.push(file.path);
+          } else {
+            initialJS.push(file.path);
+          }
+        } else {
+          if (file.path.endsWith('.css')) {
+            asyncCSS.push(file.path);
+          } else {
+            asyncJS.push(file.path);
+          }
+        }
+
+        const relatedLICENSE = licenseMap.get(file.path);
+
+        if (relatedLICENSE) {
+          assets.add(relatedLICENSE);
+        }
+
+        for (const auxiliaryFile of file.chunk!.auxiliaryFiles) {
+          assets.add(auxiliaryFile);
+        }
+      }
+
+      entries[name] = {
+        initial: {
+          js: initialJS,
+          css: initialCSS,
+        },
+        async: {
+          js: asyncJS,
+          css: asyncCSS,
+        },
+        assets: Array.from(assets),
+        html: files.find((f) => f.name === htmlPaths[name])?.path,
+      };
+    }
+
+    return {
+      allFiles,
+      entries,
+    };
+  };
+
+export const pluginManifest = (): RsbuildPlugin => ({
+  name: 'rsbuild:manifest',
+
+  setup(api) {
+    api.modifyBundlerChain(async (chain, { target, CHAIN_ID }) => {
+      const htmlPaths = api.getHTMLPaths();
+
+      const {
+        output: { manifest },
+      } = api.getNormalizedConfig();
+
+      if (manifest === false) {
+        return;
+      }
+
+      const fileName =
+        typeof manifest === 'string'
```

Comment:
Can manifest string be a file path?

Reply:
yes, manifest `string` can be a file name or a file path